cloud_storage/cloud_storage/overrides/file.py

```diff
@@ -50,17 +52,74 @@ def on_trash(self) -> None:
 				_("This file is attached to a submitted document and cannot be deleted"),
 				frappe.PermissionError,
 			)
-
+		if len(self.file_association) > 1:
+			frappe.throw(
+				_("This file is attached to multiple documents and cannot be deleted"),
+				frappe.PermissionError,
+			)
 		super().on_trash()
 
+	def associate_files(self) -> File:
+		if not self.content_hash and "/api/method/retrieve" in self.file_url:  # type: ignore
+			associated_doc = frappe.get_value("File", {"file_url": self.file_url}, "name")
+		else:
```

Comment:
So it'll enter this block on any one of the following:
- the file document has a content hash set
- the file URL is not the string the cloud storage API expects

Should both cases be handled this way?

Reply:
Yes:
The first case covers the second upload of a file (attachment, jpeg, etc)
The second case covers when a File (doctype) is selected from the library.